File: ShareButton/app/src/main/java/com/eypcnn/sharebutton/MainActivity.java
package com.eypcnn.sharebutton;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    Button eyp;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        eyp=(Button)findViewById(R.id.paylas);
        eyp.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent myIntent = new Intent((Intent.ACTION_SEND));
                myIntent.setType("text/plain");
                String shareBody = "TestApp";
                String shareSub = "EypCnn";
                myIntent.putExtra(Intent.EXTRA_SUBJECT, shareBody);
                myIntent.putExtra(Intent.EXTRA_TEXT, shareSub);
                startActivity(Intent.createChooser(myIntent, "Şununla Paylaş"));
            }
        });
    }
}

File: OpenActivityKotlin/app/src/main/java/com/eypcnn/openactivitykotlin/MainActivity.kt
package com.eypcnn.openactivitykotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val next = findViewById<View>(R.id.nextButton) as Button
        next.setOnClickListener { startActivity(Intent(this@MainActivity, MainActivity2::class.java)) }

    }
}